beacon_node/http_api/src/publish_blocks.rs

```diff
@@ -168,12 +166,29 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlockConten
         VariableList::from(blobs)
     });
 
+    // TODO: evaluate whether data columns should be constructed in `into_gossip_verified_block`
```

Comment:
Yeah I think we should gossip validate the data columns before publishing them

Reply:
definitely, but what I meant by the `TODO` was a question of whether we should add the data columns to the [`GossipVerifiedBlockContents`](https://github.com/sigp/lighthouse/blob/ae470d70041339f00de5d404ff75bcc3e0cf31eb/beacon_node/beacon_chain/src/block_verification_types.rs#L323).

Reply:
Ah right, yeah I think so. Look like the type alias and impls are mainly for code organisation and grouping error handling together. So perhaps we can just construct the `GossipVerifiedDataColumn` here:

https://github.com/sigp/lighthouse/blob/ae470d70041339f00de5d404ff75bcc3e0cf31eb/beacon_node/beacon_chain/src/block_verification.rs#L717-L743